lib/journey.ex

```diff
@@ -89,8 +89,9 @@ defmodule Journey do
   defp call(func, %__MODULE__{} = journey, _) when is_function(func, 1) do
     try do
       func.(journey)
-    rescue
-      error -> {:error, error}
+    catch
+      :error,reason -> {:error, reason}
+      error,reason -> {:error, {error, reason}}
```

Comment:
```suggestion
    catch
      :error, reason -> {:error, reason}
      error, reason -> {:error, {error, reason}}
      error -> {:error, error}
```

Shouldn't we keep the match for one error message? In case the journey returns such format.

Reply:
> Shouldn't we keep the match for one error message? In case the journey execution returns such format.

As stated by the tests and the tech reference, catch/2 is always being able to match any kind of exception (raise/1, raise/2, throw/1, exit/1, error/1), so this last clause would be unreachable.

Reply:
More info regarding compatible error handlings, but basically `catch/2` will handle all:
https://github.com/TylerPachal/elixir-error-handling-blogpost

Reply:
Great, that makes sense. Otherwise, any error that is not catchable will be returned as usual, I imagine.